src/ray/raylet/worker_pool.cc

```diff
@@ -1352,13 +1352,15 @@ void WorkerPool::PrestartWorkers(const TaskSpecification &task_spec,
   }
 }
 
-bool WorkerPool::DisconnectWorker(const std::shared_ptr<WorkerInterface> &worker,
+void WorkerPool::DisconnectWorker(const std::shared_ptr<WorkerInterface> &worker,
                                   rpc::WorkerExitType disconnect_type) {
   MarkPortAsFree(worker->AssignedPort());
   auto &state = GetStateForLanguage(worker->GetLanguage());
   auto it = state.worker_processes.find(worker->GetStartupToken());
   if (it != state.worker_processes.end()) {
-    it->second.alive_started_workers.erase(worker);
+    if (!RemoveWorker(it->second.alive_started_workers, worker)) {
```

Comment:
Can you add a comment on this block? I am a little bit confused here (isn't it supposed to be `if (RemoveWorker)`)?

Reply:
I assume it means "if the worker is not started yet, it means it is starting, so we should subtract it?"